MAUI/DataGrid/Column-types.md

```diff
@@ -521,6 +580,227 @@ dataGrid.Columns.Add(dateColumn);
+
+![DataGrid with editing in comboBox column](Images\column-types\maui-datagrid-comboBox-column.png)
+
+### Collection of primitive types
+
+To display the collection of items in the comboBox drop down, create a [DataGridComboBoxColumn]() and set its [DataGridComboBoxColumn.ItemsSource]() property to a simple collection. 
+
+To load the `DataGridComboBoxColumn` with a simple string collection, follow the code example:
+
+{% tabs %}
+{% highlight xaml %}
+    <ContentPage.BindingContext>
+        <local:ViewModel x:Name="viewModel" />
+    </ContentPage.BindingContext>
+
+    <sfGrid:SfDataGrid x:Name="dataGrid"
+                       ItemsSource="{Binding OrdersInfo}">
+        <sfGrid:SfDataGrid.Columns>
+            <sfgrid:DataGridComboBoxColumn BindingContext="{x:Reference viewModel}"
+                                           HeaderText="Name"
+                                           ItemsSource="{Binding CustomerNames}"
+                                           MappingName="DealerName" />
+        </sfGrid:SfDataGrid.Columns>
+    </sfGrid:SfDataGrid>
+{% endhighlight %}
+
+{% highlight c# %}
+dataGrid = new SfDataGrid();
+DataGridComboBoxColumn comboBoxColumn = new DataGridComboBoxColumn()
+{
+    BindingContext = viewModel,
+    MappingName = "DealerName",
+    ItemsSource = viewModel.CustomerNames,
+    HeaderText = "Name"
+
+};
+dataGrid.Columns.Add(comboBoxColumn);
+{% endhighlight %}
+{% endtabs %}
+
+{% highlight c# %}
+// ViewModel class
+public class ViewModel
+{
+    public ObservableCollection<string> CustomerNames { get; set; }
+
+    public ViewModel()
+    {
+        this.CustomerNames = Customers.ToObservableCollection();
+    }
+
+    internal string[] Customers = new string[] {"Adams","Crowley","Ellis","Gable","Irvine","Keefe","Mendoza","Owens","Rooney","Wadded",};
+    
+}
+{% endhighlight %}
+
+### Collection of user-defined types
+
+To display a list of user-defined items in the combo-box drop down, create a [DataGridComboBoxColumn]() and set its [DataGridComboBoxColumn.ItemsSource]() property to a user-defined collection. By default, the combo-box column will display the values from the [DataGridColumn.MappingName]() property of the column if the [DisplayMemberPath]() is not set.
+
+### Loading Different ItemSource for each row of DataGridComboBoxColumn
+
+You can load the different ItemsSource to each row of DataGridComboBoxColumn by setting [DataGridComboBoxColumn.ItemsSourceSelector]() property.
+
+### Implementing IItemsSourceSelector
+
+[DataGridComboBoxColumn.ItemsSourceSelector]() needs to implement IItemsSourceSelector interface which requires you to implement GetItemsSource method which receives the below parameters,
+
+* Record – data object associated with row.
+* Data Context – Binding context of data grid.
+
+In the below code, ItemsSource for ShipCity column returned based on ShipCountry column value using the record and Binding context of data grid passed to GetItemsSource method.
```

Comment:
[DataGridComboBoxColumn.ItemsSourceSelector]() needs to implement the IItemsSourceSelector interface, which requires you to implement the GetItemsSource method. This method receives the following parameters:

- Record: This is the data object associated with the row.
- Data Context: This is the binding context of the data grid.

In the provided code, the ItemsSource for the ShipCity column is returned based on the value of the ShipCountry column. This is done by using the record and the binding context of the data grid, which are passed to the GetItemsSource method.

Reply:
![image](https://github.com/syncfusion-content/maui-docs/assets/113881396/80f68c82-cdba-4788-9aff-aed61f8ea744)